<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="x_pos" default="0.0" />
    <arg name="y_pos" default="0.0" />
    <arg name="z_pos" default="0.0" />
    <arg name="yaw" default="0.0" />
    <arg name="gui" default="true"/>
    <arg name="world" default="empty"/>

  <rosparam  file="$(find assignment_1)/config/joint_state_controller.yaml" command="load" />
  <rosparam file="$(find assignment_1)/config/aruco_navigation.yaml" command="load" />
  <param name="robot_description" command="$(find xacro)/xacro '$(find rosbot_description)/urdf/rosbot.xacro'"/>  
 
  <node name="joint_state_controller_spawner" pkg="controller_manager" type="spawner" output="screen" args="joint_state_controller" />
  <node name="rosbot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen" args="-urdf -param robot_description -model rosbot -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -Y $(arg yaw)"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rosbot_description)/rviz/mapping.rviz"/>
  <node name="aruco_detector" pkg="assignment_1" type="aruco_detector.py" required="true"/>
  <node name="robot_control" pkg="assignment_1" type="robot_control.py" required="true" output="screen" launch-prefix="xterm -hold -e"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find assignment_1)/worlds/aruco_assignment.world"/> 
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
  </include>
  
</launch>
